unittests/linuxliketests.py

```diff
@@ -256,6 +256,36 @@ def test_pkgconfig_gen_deps(self):
             content = f.read()
             self.assertNotIn('-lstat2', content)
 
+    def test_pkgconfig_relocatable(self):
+        '''
+        Test that it generates relocatable pkgconfig when module
+        option pkgconfig.relocatable=true.
+        '''
+        testdir_rel = os.path.join(self.common_test_dir, '44 pkgconfig-gen')
+        self.init(testdir_rel, extra_args=['-Dpkgconfig.relocatable=true'])
+
+        def check_pcfile(name, *, relocatable, levels=2):
+            with open(os.path.join(self.privatedir, name), encoding='utf-8') as f:
+                pcfile = f.read()
+                (self.assertIn if relocatable else self.assertNotIn)(
+                    f'prefix={os.path.join("${pcfiledir}", *(".." for _ in range(levels)))}\n',
```

Comment:
```suggestion
                    f'prefix={os.path.join("${pcfiledir}", *([".."] *levels))}\n',
```
is much faster, per timeit.

Aside: I don't like the use of extremely complex expressions inside f-strings as it is extremely hard to read.

Reply:
I'll move that expression outside the f-strings.

Reply:
In the latest commit I moved the expression outside the f-string and switched to `['..'] * levels`. Hopefully it's a bit more readable now.